numpy/f2py/__init__.py

```diff
@@ -59,10 +62,32 @@ def compile(source,
         f.write(source)
         f.flush()
 
-        args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
-        c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
-        c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        args = ['-c', '-m', modulename, f.name]
+
+        if os.name == 'posix':
+            posix = True
+        else:
+            posix = False
+
+        extra_args = shlex.split(extra_args, posix=posix)
+
+        if extra_args is not None:
+            args.extend(extra_args)
+
+        c = [sys.executable,
+             '-c',
+             'import numpy.f2py as f2py2e;f2py2e.main()'] + args
```

Comment:
Seems bizarre to me that we go through subprocess to do this when we could just call the `main` function that is in scope here...

But I guess that's not the aim of this patch - something to consider in another PR